src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupDeltaDDC.java

```diff
@@ -19,11 +19,271 @@
 
 package org.apache.sysds.runtime.compress.colgroup;
 
+
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.sysds.runtime.compress.colgroup.dictionary.ADictionary;
+import org.apache.sysds.runtime.compress.colgroup.dictionary.DeltaDictionary;
+import org.apache.sysds.runtime.compress.colgroup.indexes.IColIndex;
+import org.apache.sysds.runtime.compress.colgroup.mapping.AMapToData;
+import org.apache.sysds.runtime.compress.colgroup.scheme.ICLAScheme;
+import org.apache.sysds.runtime.compress.cost.ComputationCostEstimator;
+import org.apache.sysds.runtime.data.DenseBlock;
+import org.apache.sysds.runtime.data.SparseBlock;
+import org.apache.sysds.runtime.functionobjects.Builtin;
+import org.apache.sysds.runtime.instructions.cp.CM_COV_Object;
+import org.apache.sysds.runtime.matrix.data.MatrixBlock;
+import org.apache.sysds.runtime.matrix.operators.BinaryOperator;
+import org.apache.sysds.runtime.matrix.operators.CMOperator;
+import org.apache.sysds.runtime.matrix.operators.ScalarOperator;
+import org.apache.sysds.runtime.matrix.operators.UnaryOperator;
+
 /**
  * Class to encapsulate information about a column group that is first delta encoded then encoded with dense dictionary
  * encoding (DeltaDDC).
  */
-public class ColGroupDeltaDDC  { // extends ColGroupDDC
+public class ColGroupDeltaDDC extends AColGroupCompressed implements AMapToDataGroup  {
+
+    private static final long serialVersionUID = -1045556313148564147L;
+    private AMapToData _data;
+    private DeltaDictionary _dict;
+    //TODO Confirm if casting from ADictionary to DeltaDictionary is reasonable
+    private ColGroupDeltaDDC(IColIndex colIndexes, ADictionary dict, AMapToData data) {
+        super(colIndexes);
+        _data = data;
+        _dict = (DeltaDictionary) dict;
+    }
+
+    //TODO Confirm if other parameters for constructors are required, such as cachedCounts
+    public static AColGroup create(IColIndex colIndexes, ADictionary dict, AMapToData data) {
+        if(data.getUnique() == 1)
+            return ColGroupConst.create(colIndexes, dict);
+        else if(dict == null)
+            return new ColGroupEmpty(colIndexes);
```

Comment:
not valid.
on the other hand if you observe that all instances point to a dictionary entry with only 0, then it is valid. 

consider the case that all deltas throughout the matrix is 0, then you know that it is equivalent to an empty matrix.

Reply:
Should I also considerer a case where every element in a column is the same? In a column (-3,-3,-3,-3), the first delta would  be -3 and all other deltas 0. In that case I would check if the first delta value is 0 and all other deltas are the same, then it would be possible to create a constant column group? Or have I misunderstunt the definition of a constant column group? 

 if (dict.getValues().length == 1 && dict.getValue(0)==0), that would represent not an empty Matrix, but a Matrix that only contains zeros, therefore, my recent code is invalid I suppose. 

The definition of an empty matrix in this context is a matrix with only zeroes or is it a matrix with at least one dimension that is equal to zero?

Reply:
yes, if the column can be encoded as a constant or an empty column group the return should be an empty column group. 
The case you write is exactly the case we need to support.

But !

While it would be good that we return the specialized instance of const and empty column groups,  you do not have to add it if it is confusing, just make sure that the thing that is returned from the construct is correct, or throw an error if incorrect.